### PR TITLE
Deprecate apachepulsar/pulsar-grafana docker image

### PR DESCRIPTION
**Pull request by michaeljmarshall**
### Motivation

Clean up our `docker` directory. We currently produce a very outdated `pulsar-grafana` docker image with each release. The primary reason we're producing this image is to package grafana dashboards. Grafana allows easy import through the UI, or if you're running on Kubernetes, you can provide the dashboards as config maps. Given how old these files are, it looks like the dashboards do not receive much attention.

### Modifications

* Move the grafana dashboards to a top level directory. (We could also remove them completely if we don't intend to maintain them.)
* Remove the mvn build script for the `pulsar-grafana` docker image.

### Verifying this change

Once tests pass, I'll send a note to the dev mailing list to make sure we get the right visibility on this change before merging it.

### Does this pull request potentially affect one of the following parts:

This PR deprecates our custom grafana docker image.

### Documentation
  
- [x] `doc` 
  
I updated the latest docs with this change.

### CODE REVIEW

**Comment by michaeljmarshall** [comment]
Note that we currently use this docker image in a dcos example here https://github.com/apache/pulsar/blob/c5d7a84c8e5c27e48022df8c7082496840cd3be9/deployment/dcos/PulsarGroups.json#L310

I think it might be worth removing the DCOS example all together, since it hasn't been updated in 4 years.

File: site2/docs/deploy-monitoring.md [hunk 1]
```diff
@@ -108,15 +108,7 @@ The per-topic dashboard instructions are available at [Pulsar manager](administr
...
-When you deploy Pulsar on Kubernetes, a `pulsar-grafana` Docker image is enabled by default. You can use the docker image with the principal dashboards.
-
-Enter the command below to use the dashboard manually:
-
-```shell
-docker run -p3000:3000 \
-        -e PROMETHEUS_URL=http://$PROMETHEUS_HOST:9090/ \
-        apachepulsar/pulsar-grafana:latest
-```
+When you deploy Pulsar on Kubernetes with the Pulsar Helm Chart, a `pulsar-grafana` Docker image is enabled by default. You can use the docker image with the principal dashboards.
```
Comment (Anonymitaet) [inline site2/docs/deploy-monitoring.md:111]: @freeznet can you review this PR from the technical perspective?

**Comment by freeznet** [comment]
Can you start a mail thread in the pulsar maillist so that users and devs can get updates about this PR?

**Comment by michaeljmarshall** [comment]
> Once tests pass, I'll send a note to the dev mailing list to make sure we get the right visibility on this change before merging it.

@freeznet - yes, my plan is to send a note once I get tests passing.

**Review by merlimat — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**